modules/core/src/main/java/org/apache/ignite/internal/properties/IgniteProductVersion.java

```diff
@@ -47,11 +48,15 @@ public class IgniteProductVersion implements Serializable {
     /** Flag indicating if this is a snapshot release. */
     private final boolean isSnapshot;
 
-    private IgniteProductVersion(byte major, byte minor, byte maintenance, boolean isSnapshot) {
+    /** Alpha version part or an empty string if this is not an alpha release. */
+    private final String alphaVersion;
```

Comment:
What if at some point we switch to beta versions? Should we generalize over 'alpha|beta|rc|...'?

Reply:
Add https://issues.apache.org/jira/browse/IGNITE-17146